pkg/vmprovider/providers/vsphere/session/session_vm_customization.go

```diff
@@ -261,6 +301,12 @@ func customizeCloudInit(
 	}
 
 	if err != nil {
+		errWithReason := err.(ErrWithReason)
```

Comment:
This was supposed to be `errWithReason := err.(*ErrWithReason)`
Will try to add unit tests for this one.